File: TOI-Zero_68/A2/031/031.py
n = int(input())
vec = [[], []]
vec[0] = input().strip().split()
vec[1] = input().strip().split()

q = int(input())
for _ in range(q):
    a, b, c = input().split()
    a, b = int(a), int(b)
    vec[a - 1][b] = c

print(" ".join(vec[0]))
print(" ".join(vec[1]))

cnt = 0
for i in range(n):
    s = vec[0][i] + vec[1][i]
    if s in ["AT", "TA", "CG", "GC"]:
        cnt += 1

print(n - cnt)

File: TOI-Zero_68/A2/031/031.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
	int n;
	cin>>n;
	vector<char> vec[2];
	vec[0].resize(n);
	vec[1].resize(n);
	for(int i=0;i<n;i++){
		cin>>vec[0][i];
	}
	for(int i=0;i<n;i++){
		cin>>vec[1][i];
	}
	int q;
	cin>>q;
	while(q--){
		int a,b;
		char c;
		cin>>a>>b>>c;
		vec[a-1][b]=c;
	}
	for(int i=0;i<n;i++){
		cout<<vec[0][i]<<" ";
	}
	cout<<"\n";
	for(int i=0;i<n;i++){
		cout<<vec[1][i]<<" ";
	}
	cout<<"\n";
	int cnt=0;
	for(int i=0;i<n;i++){
		string s="";
		s+=vec[0][i];
		s+=vec[1][i];
		if(s=="AT"||s=="TA"||s=="CG"||s=="GC") cnt++;
	}
	cout<<n-cnt;
	return 0;
}
